clis/teliod/src/daemon.rs

```diff
@@ -0,0 +1,64 @@
+//! Code for general daemon handling utilities and abstracting away related dependencies.
+
+use std::{
+    fs::{self, File},
+    io::{Error, ErrorKind, Result, Write},
+    path::{Path, PathBuf},
+    process::id,
+};
+
+/// Daemon name, used for naming daemon files
+const DAEMON_NAME: &str = "teliod";
+
+/// Struct for packaging together the functions for managing the daemon.
+/// Currently does not need any members, since most of the data is stored
+/// in files anyway.
+pub struct TeliodDaemon {
+    pid_path: PathBuf,
+}
+
+impl TeliodDaemon {
+    /// Initializes all of the needed daemon stuff, currently just creates a PID file
+    pub fn new() -> Result<Self> {
```

Comment:
I think pid-file is not used anywhere anymore

Reply:
I thought the deamon just should have some pidfile, is used in the test for killing and maybe for such purposes but I can probably use `ps aux` instead. Idk, I can remove it.